.github/code/does_dataset_have_sessions.py

```diff
@@ -0,0 +1,39 @@
+from pathlib import Path
+import sys
+import argparse
+
+
+def does_dataset_have_sessions(dataset_id: str, bids_path: Path):
+    """
+    Exit with code 0 if any sessions exist, 1 otherwise.
+    Warn if some subjects have sessions and others do not.
+    """
+    subjects_with_sessions = []
+    subjects_without_sessions = []
+    for sub_dir in bids_path.glob('sub-*'):
+        if sub_dir.is_dir():
+            sub_has_any_ses = any(ses_dir.is_dir() for ses_dir in sub_dir.glob('ses-*'))
+            if sub_has_any_ses:
+                subjects_with_sessions.append(sub_dir.name)
+            else:
+                subjects_without_sessions.append(sub_dir.name)
+
+    if subjects_with_sessions and subjects_without_sessions:
+        print(
+            f"{dataset_id}: WARNING Dataset has a session layer but "
+            f"{len(subjects_without_sessions)} subject(s) lack session subdirectories: "
+            f"{subjects_without_sessions}"
+        )
```

Comment:
What should we do in such a case? Do we expect such a case? afaik, this would not be allowed under BIDS.

Reply:
Good point. This is probably being over-careful, but having tested it locally it does look like a dataset containing some subjects with `ses-` directories and some without (e.g., having data directly under the `sub-` folder) still passes BIDS validation, at least using `pybids`.

I think we can try to catch these cases more to report back to the OpenNeuro folks than anything else. In terms of what the script should do, I think it'd make sense to exit and skip a dataset with inconsistent sessions, since the nipoppy tracking might produce weird results (or fail completely). At the moment it will still try to run the tracking - will fix that.